lib/cstruct.mli

```diff
@@ -221,6 +221,15 @@ val of_bytes: ?allocator:(int -> t) -> Bytes.t -> t
     with the underlying buffer allocated by [alloc]. If [allocator] is not
     provided, [create] is used. *)
 
+val of_hex: string -> t
+(** [of_hex str] is the cstruct of [str].  Every pair of hex-encoded characters
+    from [str] are converted to a single byte in the cstruct.  The resulting
+    cstruct is exactly half the size of [str].  Whitespaces (space, newline,
```

Comment:
When the [str] len is odd, that is not the case :) Maybe *The length of the resulting cstruct is half of the length of the hex characters in [str]*? (While technically the underlying buffer is longer (and contains unsanitized heap memory), you do return a substring for which it *is* true)

Reply:
reworded